src/common/src/hash/key.rs

```diff
@@ -164,16 +165,17 @@ pub trait HashKey:
 /// | (1, null) | (1, null) | true |
 /// | (null, 1) | (null, 1) | false |
 /// | (null, null) | (null, null) | false |
-pub struct JoinHashKey<'a, K> {
+#[derive(Clone)]
+pub struct JoinHashKey<K> {
     key: K,
-    null_matched: &'a FixedBitSet,
+    null_matched: Arc<FixedBitSet>,
```

Comment:
Is this `null_matched` identical for all the entries in a hash table? If so, can we eliminate it from the key?

Reply:
Yes, they are identical. Currently, `null_matched` is mostly used to overwrite `PartialEq` trait.
```rust
impl<K: HashKey> PartialEq for JoinHashKey<K> {
    fn eq(&self, other: &Self) -> bool {
        self.key == other.key && self.key.null_bitmap().is_subset(&self.null_matched)
    }
}
```

Reply:
One possible alternative way is to make the `self.key.null_bitmap().is_subset(&self.null_matched)` to the join executor. If it's false, then no matched data in a hash map.

Reply:
Agree. Let me do some refactor.

Reply:
Please take a look again @yuhao-su and I also remove `JoinHashKey` from `BatchHashJoinExecutor`